src/lib/stores/preferences.ts

```diff
@@ -44,6 +72,30 @@ function createPreferences() {
         }
     });
 
+    /**
+     * Update the local store and then synchronizes them on user prefs.
+     */
+    function updateAndSync(callback: (prefs: PreferencesStore) => void) {
+        let oldPrefsSnapshot: string;
+        let newPrefsSnapshot: PreferencesStore;
+
+        update((currentPrefs) => {
+            oldPrefsSnapshot = JSON.stringify(currentPrefs);
+            callback(currentPrefs);
+            newPrefsSnapshot = currentPrefs;
+            return currentPrefs;
+        });
+
+        // Skip API if no changes (sufficient for simple objects).
+        // The key order seemed to be maintained during local tests.
+        if (oldPrefsSnapshot === JSON.stringify(newPrefsSnapshot)) {
+            return;
+        }
+
+        // sync the preferences.
+        updateConsolePreferences(newPrefsSnapshot).then();
+    }
```

Comment:
not sure what this function is supposed to do? :-D

Reply:
I dont really get this part `updateConsolePreferences(newPrefsSnapshot).then();`